lib/supavisor/client_handler.ex

```diff
@@ -828,4 +819,20 @@ defmodule Supavisor.ClientHandler do
   end
 
   defp handle_prepared_statements(_, _, _), do: nil
+
+  @spec handle_actions(map) :: [{:timeout, non_neg_integer, atom}]
+  defp handle_actions(data) do
+    Enum.reduce(data, [], fn
+      {:heartbeat_interval, v} = t, acc when v > 0 ->
+        Logger.debug("Call timeout #{inspect(t)}")
+        [timeout_check(:heartbeat_check, v) | acc]
+
+      {:idle_timeout, v} = t, acc when v > 0 ->
+        Logger.debug("Call timeout #{inspect(t)}")
+        [timeout_check(:idle_terminate, v) | acc]
+
+      _, acc ->
+        acc
+    end)
```

Comment:
Just a tiny suggestion that you can write code that has to filter and map as a `flat_map`. :)

```suggestion
    Enum.flat_map(data, fn
      {:heartbeat_interval, v} = t when v > 0 ->
        Logger.debug("Call timeout #{inspect(t)}")
        [timeout_check(:heartbeat_check, v)]

      {:idle_timeout, v} = t when v > 0 ->
        Logger.debug("Call timeout #{inspect(t)}")
        [timeout_check(:idle_terminate, v)]

      _ ->
        []
    end)
```

But if you prefer reduce, feel free to stick with it!